### ISSUE TITLE
dbcrossbar doesn't support --if-exists=error for bigquery

### ISSUE THREAD

**Issue by skelliest** [dbcrossbar/dbcrossbar#119]
Attempting to use dbcrossbar to copy a table to bigquery, I get this error

```
$ dbcrossbar cp csv:tmp.csv bigquery:production-237317:tmp.sk1 --temporary=gs://fdy-production-tmp/x
CommandException: 1 files/objects could not be removed.
Copying from <STDIN>...
/ [1 files][    0.0 B/    0.0 B]                                                
Operation completed over 1 objects.                                              
Error: this driver does not support --if-exists=error
```

1. I never set --if-exists, so it is bad behavior to require an optional parameter
2. BigQuery actually does support this functionality, unless I'm not understanding the purpose of the flag. According to the WriteDisposition documentation (https://cloud.google.com/bigquery/docs/reference/auditlogs/rest/Shared.Types/WriteDisposition):

> WriteDisposition
> Describes whether a mutation to a table should overwrite or append.

> Enums
> WRITE_DISPOSITION_UNSPECIFIED	Unknown.
> WRITE_EMPTY	This job should only be writing to empty tables.
> WRITE_TRUNCATE	This job will truncate table data and write from the beginning.
> WRITE_APPEND	This job will append to a table.

And at https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.job.WriteDisposition.html#google.cloud.bigquery.job.WriteDisposition:

> WRITE_EMPTY = 'WRITE_EMPTY'
> If the table already exists and contains data, a ‘duplicate’ error is returned in the job result.

--if-exists=error is the preferred functionality for DS.

**Comment by skelliest** [comment]
@emk Thoughts?

**Comment by emk** [comment]
Hmm, if I remember correctly, we didn't support `--if-exits=error` because the `bq` command-line tool originally didn't support it, either.

Now that we've stopped using `bq`, and we're using the native BigQuery APIs, I _think_ we can do this. I've added this to the [v0.4.0 milestone](https://github.com/dbcrossbar/dbcrossbar/milestone/5), which is currently blocked on me finding time to speed up Google Cloud Storage uploads.
